File: flink-sample-java/src/main/java/RollingSum.java
import org.apache.flink.api.java.tuple.Tuple3;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

public class RollingSum {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        DataStream<Tuple3<Integer, Integer, Integer>> inputStream = env
                .fromElements(Tuple3.of(1,2,2), Tuple3.of(2,3,1), Tuple3.of(2,2,4), Tuple3.of(1,5,3));
        DataStream<Tuple3<Integer, Integer, Integer>> resultStream = inputStream.keyBy(0).sum(1);
        resultStream.print();
        env.execute();
    }
}

File: flink-datastream-1.7.1/src/main/scala/scalalang/RollingSum.scala
package scalalang

import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment
import org.apache.flink.streaming.api.scala._

object RollingSum {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    val inputStream = env.fromElements((1,2,2),(2,3,1),(2,2,4),(1,5,3))
    val resultStream = inputStream.keyBy(0).sum(1)
    resultStream.print()
    env.execute()
  }
}
